lib/mongodb.js

```diff
@@ -734,8 +734,13 @@ MongoDB.prototype.buildSort = function(model, order) {
 };
 
 function buildNearFilter(query, near) {
+  var node = query.where;
+  near.key.forEach(function(key) {
+    node = node[key];
+  });
+
```

Comment:
```+  near.key.forEach(function(key) {
+    node = node[key];
+  });```

Your code here will loop through all the keys and node will be left assigned to the last key it encountered.